src/Geometry/CSG/CSG.jl

```diff
@@ -115,69 +125,94 @@ end
 
 Create a leaf node from a parametric surface with a given transform.
 """
-function leaf(surf::S, transform::Transform{T} = identitytransform(T))::CSGGenerator{T} where {T<:Real,S<:ParametricSurface{T}}
+function leaf(surf::ParametricSurface{T}, transform::Transform{T} = identitytransform(T)) where {T<:Real}
     return CSGGenerator{T}((parenttransform) -> LeafNode(surf, parenttransform * transform))
 end
 """
     leaf(surf::CSGGenerator{T}, transform::Transform{T} = identitytransform(T)) -> CSGGenerator{T}
 
-Create a (pseudo) leaf node from another CSGGenerator, this is useful if you want multiple copies of a premade CSG structure with different transforms, for example in an MLA.
+Create a (pseudo) leaf node from another CSGGenerator, this is useful if you want multiple copies of a premade CSG
+structure with different transforms, for example in an MLA.
 """
-function leaf(n::CSGGenerator{T}, transform::Transform{T} = identitytransform(T))::CSGGenerator{T} where {T<:Real}
+function leaf(n::CSGGenerator{T}, transform::Transform{T} = identitytransform(T)) where {T<:Real}
```

Comment:
yeah exactly as Brian suggests:
```suggestion
function transform(n::CSGGenerator{T}, transform::Transform{T}) where {T<:Real}
```